### PR TITLE
allow vault passwords with SCM inventories

### PR DESCRIPTION
**Pull request by AlanCoding**
Connect #223

Note, you need Ansible 2.4 to use this, unless we added support in the backport for it.

### CODE REVIEW

**Comment by matburt** [comment]
Is there anything that needs to be changed about the UI to allow selecting a vault password in the right place?

**Comment by AlanCoding** [comment]
Yes, it would need to be added as a selector filtered to vault type credentials in the Project-based inventory source edit screen.

File: awx/api/serializers.py [hunk 1]
```diff
@@ -1533,7 +1533,8 @@ def get_related(self, obj):
 class InventorySourceOptionsSerializer(BaseSerializer):
 
     class Meta:
-        fields = ('*', 'source', 'source_path', 'source_script', 'source_vars', 'credential',
+        fields = ('*', 'source', 'source_path', 'source_script', 'source_vars',
+                  'credential', 'vault_credential',
```
Comment (AlanCoding) [inline awx/api/serializers.py:1527]: Actually, this will need to copy the validation used in job templates.

Comment (matburt) [reply awx/api/serializers.py:1527]: 👍

Comment (AlanCoding) [reply awx/api/serializers.py:1527]: Added that, but by way of the `clean_vault_credential` on model.

**Review by matburt — APPROVED** [review]
Make sure we raise this with ui/qe to get this in front of folks after this merges. Might need some issues to track.

**Comment by AlanCoding** [comment]
Added notes in the CHANGELOG and the docs for SCM inventory

**Comment by wenottingham** [comment]
Just to note: this does mean that vaulted-in-SCM things will be in plaintext in the Tower API/UI.

**Comment by AlanCoding** [comment]
Closing with intent to re-target against branch for https://github.com/ansible/awx/issues/352

**Comment by Elyytscha** [comment]
is this merged already? because i have to set vault credentials for inventory source from scm (git)

our ansible.cfg is located in our ansible repos so the inventory sync always fails with
`ansible-inventory [core 2.11.2.post0] 
  config file = /runner/project/ansible.cfg
  configured module search path = ['/home/runner/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/site-packages/ansible
  ansible collection location = /runner/requirements_collections:/home/runner/.ansible/collections:/usr/share/ansible/collections:/usr/share/automation-controller/collections
  executable location = /usr/local/bin/ansible-inventory
  python version = 3.8.6 (default, Jan 29 2021, 17:38:16) [GCC 8.4.1 20200928 (Red Hat 8.4.1-1)]
  jinja version = 2.10.3
  libyaml = True
Using /runner/project/ansible.cfg as config file
ERROR! The vault password file /home/runner/.vault_pass was not found`

**Pull request closed without merging**